lib/internal/modules/esm/loader.js

```diff
@@ -163,72 +179,83 @@ class Loader {
     return module.getNamespace();
   }
 
+  async importLoader(specifier) {
+    if (cwd === undefined) {
+      try {
+        // `process.cwd()` can fail.
+        cwd = process.cwd() + '/';
+      } catch {
+        cwd = 'file:///';
+      }
+      cwd = pathToFileURL(cwd).href;
+    }
+
+    const { url } = await defaultResolve(specifier, cwd,
+                                         { conditions: DEFAULT_CONDITIONS });
+    const { format } = await defaultGetFormat(url, {});
+
+    // !!! CRITICAL SECTION !!!
+    // NO AWAIT OPS BETWEEN HERE AND SETTING JOB IN MODULE MAP
```

Comment:
Can you please explain why this rule is here? Ideally as a comment in the code explaining what bad things will happen if one were to disregard the warning and add an `await` in the critical section.